incubator/hnc/internal/reconcilers/object.go

```diff
@@ -431,13 +432,11 @@ func (r *ObjectReconciler) syncSource(ctx context.Context, log logr.Logger, src
 	ns := r.Forest.Get(src.GetNamespace())
 	ns.SetSourceObject(src.DeepCopy())
 
-	if !r.shouldPropagateSource(log, src) {
-		// If an object was *previously* propagated by HNC, it will already be in the forest, and all
-		// the propagated copies will think they can stick around too. If, for some reason, we _now_
-		// want to stop propagating it (e.g. because we've changed the sync mode in the HNCConfig
-		// object), we need to delete it from the forest and enqueue all its propagated copies. The
-		// propagated copies will then see that that the source is missing from the forest, and delete
-		// themselves.
+	// We are only checking if the source itself is propagatable, but not to a specific destination.
```

Comment:
Hmm, I'm wondering if we can simplify the rest of this function. Right now it's:

```
if !should propagate source {
  sync unpropagated source()
  // this does an hncconfig reconcile and enqueues all descendants
  return
}
hncconfig reconcile
enqueue descendants
```

But both the `if` and (implied) `else` clause here are the same! So we can just get rid of the call to `shouldPropagateSource`, and since (I think) this is the only place where we call that function with the empty string as the namespace, we can remove the logic to handle "all namespaces" too. Does that look correct to you?

Reply:
yea that looks right, done